lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -0,0 +1,220 @@
+package Mojo::WebSocketProxy::Dispatcher;
+
+use Mojo::Base 'Mojolicious::Controller';
+use Mojo::WebSocketProxy::Dispatcher::Parser;
+use Mojo::WebSocketProxy::Dispatcher::Config;
+use Mojo::WebSocketProxy::CallingEngine;
+
+use Time::Out qw(timeout);
+
+sub ok {
+    my $c      = shift;
+    my $source = 1;       # check http origin here
+    $c->stash(source => $source);
+    return 1;
+}
+
+sub open_connection {
+    my ($c) = @_;
+
+    my $log = $c->app->log;
+    $log->debug("opening a websocket for " . $c->tx->remote_address);
+
+    # Enable permessage-deflate
+    $c->tx->with_compression;
+
+    my $config = Mojo::WebSocketProxy::Dispatcher::Config->new->{config};
+
+    Mojo::IOLoop->singleton->stream($c->tx->connection)->timeout($config->{stream_timeout}) if $config->{stream_timeout};
+    Mojo::IOLoop->singleton->max_connections($config->{max_connections}) if $config->{max_connections};
+
+    $config->{opened_connection}->($c) if $config->{opened_connection} and ref($config->{opened_connection}) eq 'CODE';
```

Comment:
I think check `ref() eq 'CODE'` is too late here (every time, at runtime). That should be checked in plugin configuration time